### PR TITLE
Updated readme with a prehandler hook example

### PR DESCRIPTION
**Pull request by cemremengu**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -29,6 +29,34 @@ fastify.listen(3000, err => {
...
+```js
+const fastify = require('fastify')
+fastify.register(require('fastify-jwt'), { 
+  secret: 'supersecret' 
+})
+
+fastify.addHook("preHandler", async (request, reply, done) => {
+  try {
+    await request.jwtVerify()
+    done()
```
[the remark this answers is not in the document]
Comment (cemremengu) [reply README.md]: Sorry newbie here :) Should it `return request.jwtVerify()` or just `await request.jwtVerify()` ?

[the remark this answers is not in the document]
Comment (cemremengu) [reply README.md]: Assuming the latter 😄

**Comment by cemremengu** [comment]
I will update examples tomorrow with decorators to show verification for selected routes using beforeHandler

File: README.md [hunk 2]
```diff
@@ -29,6 +29,71 @@ fastify.listen(3000, err => {
...
 
+For verifying & accessing the decoded token inside your services, you can use a global `preHandler` hook to define the verification process like so:
+
+```js
+const fastify = require('fastify')
+fastify.register(require('fastify-jwt'), { 
+  secret: 'supersecret' 
+})
+
+fastify.addHook("preHandler", async (request, reply, done) => {
```
Comment (delvedor) [inline README.md]: You don't need the `done` callback in an *async* hook :)
Same below.

File: README.md [hunk 3]
```diff
@@ -29,6 +29,71 @@ fastify.listen(3000, err => {
...
+  } catch (err) {
+    reply.send(err)
+  }
+})
+```
+
+Aftewards, just use `request.user` in order to retrieve the user information:
+
+```js
+module.exports = async function(fastify, opts, next) {
```
Comment (mcollina) [inline README.md]: next is not needed here.

File: README.md [hunk 4]
```diff
@@ -29,6 +29,71 @@ fastify.listen(3000, err => {
...
+      reply.send(err)
+    }
+  })
+})
+```
+
+Then use the `beforeHandler` of a route to protect it & access the user information inside:
+
+```js
+module.exports = async function(fastify, opts, next) {
```
Comment (mcollina) [inline README.md]: next is not needed here.

**Pull request merged**